### PR TITLE
Memory pair game

### PR DESCRIPTION
**Pull request by webdevtoday**
### memory-pair-game

[Demo](https://webdevtoday.github.io/memory-pair-game/) |
[Code base](https://github.com/webdevtoday/memory-pair-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/webdevtoday/memory-pair-game/js/app.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            card.classList.remove('open');
+        });
+    }
+
+    function showScore() {
+        score.innerHTML = numberOfOpenCards;
+    }
+
+    function checkWin() {
+        if (numberOfOpenCards === 12) {
```
Comment (MaxKalashnyk) [inline submissions/webdevtoday/memory-pair-game/js/app.js:88]: Move `12` to the constants

File: submissions/webdevtoday/memory-pair-game/js/app.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        blockClicks = true;
+        el.classList.add('open');
+        openCards.push(el);
+
+        setTimeout(() => {
+            if (openCards.length === 2) checkPair();
+            showScore();
+            checkWin();
+            blockClicks = false;
+        }, 500);
```
Comment (MaxKalashnyk) [inline submissions/webdevtoday/memory-pair-game/js/app.js:111]: Move `500` time value to the constants

**Review by MaxKalashnyk — CHANGES_REQUESTED** [review]
Nice work! I'd recommend you adding some button to start a new game after finishing playing.

File: submissions/webdevtoday/memory-pair-game/css/style.css [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+.gameboard {
+    display: flex;
+    flex-wrap: wrap;
+    justify-content: center;
+    max-width: 550px;
+    margin: 0 auto;
+}
+
+.flip-container {
```
[the remark this answers is not in the document]
Comment (webdevtoday) [reply submissions/webdevtoday/memory-pair-game/css/style.css:33]: That's true, I forgot it, I did it, generally, on my phone))) I'll fix it!

File: submissions/webdevtoday/memory-pair-game/js/app.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+    const imagesTwice = doubleImages(images);
+
+    const imgDirName = 'img';
+    const imgFormat = 'png';
+    let openCards = [];
+    let blockClicks = false;
+
+    function rand(min, max) {
+        let rand = min + Math.random() * (max + 1 - min);
```
[the remark this answers is not in the document]
Comment (webdevtoday) [reply submissions/webdevtoday/memory-pair-game/js/app.js:19]: You're right! I'll fix it

File: submissions/webdevtoday/memory-pair-game/js/app.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            <div class="flip-container card">
+                <div class="flipper">
+                    <div class="front"></div>
+                    <div class="back">
+                        <img src="{{imgpath}}" alt="card image">
+                    </div>
+                </div>
+            </div>
+        `;
+        const card = cardTemplate.replace('{{imgpath}}', imgPathBuilder( randomCutImage() ));
```
[the remark this answers is not in the document]
Comment (webdevtoday) [reply submissions/webdevtoday/memory-pair-game/js/app.js:46]: Ohh, I didn't think about it, I'll change it

File: submissions/webdevtoday/memory-pair-game/js/app.js (hunk 1 above)
Comment (webdevtoday) [reply submissions/webdevtoday/memory-pair-game/js/app.js:88]: Yes, it's my favorite mistake))

File: submissions/webdevtoday/memory-pair-game/js/app.js (hunk 2 above)
Comment (webdevtoday) [reply submissions/webdevtoday/memory-pair-game/js/app.js:111]: I'll fix it

**Comment by webdevtoday** [comment]
> Nice work! I'd recommend you adding some button to start a new game after finishing playing. Look at my comments below as well

Thank you very much! I'll fix all of the problems.

**Comment by webdevtoday** [comment]
@MaxKalashnyk Hi! I fixed problems and I added New Game button.

File: submissions/webdevtoday/oop_exercise/index.js [hunk 6]
```diff
@@ -1,3 +1,75 @@
+<<<<<<< HEAD
```
Comment (MaxKalashnyk) [inline submissions/webdevtoday/oop_exercise/index.js:1]: I think you don't need this file in this PR

**Review by MaxKalashnyk — CHANGES_REQUESTED** [review]
I don't see that all of the issues are resolved. Please, fix all of them and we'll ready to approve this PR

File: submissions/webdevtoday/oop_exercise/index.js (hunk 6 above)
Comment (webdevtoday) [reply submissions/webdevtoday/oop_exercise/index.js:1]: I've lost a part of work when I made commits( and I got this file. I don't understand how to fix it(

**Comment by webdevtoday** [comment]
@MaxKalashnyk Hi! I tried to fix my problems. I revert my last commit and then I made changes in my code. Then I commit it and made steps from instruction to commit.

**Comment by MaxKalashnyk** [comment]
Your new commit affected all of the student's works in the repository. Please, fix it

**Comment by webdevtoday** [comment]
> Your new commits affected all of the student's works in the repository. Please, fix it

I don't know how to do it, could you help me, please?

**Comment by MaxKalashnyk** [comment]
Go to your `webdevtoday:memory-pair-game` branch. 
Then you can run command `git rebase -i main`. After running this command you can type `i` key in order to have an ability to edit all of your commits. Choose your last commits that broke your PR and type the `d` flag near them. It means that you'll drop them from your PR. After doing it, press `i` again. And run `:wq` for saving this configuration.
More info about the `git rebase` command you can find [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase).

Let me know if you need any help

**Comment by webdevtoday** [comment]
> Go to your `webdevtoday:memory-pair-game` branch. 
> Then you can run command `git rebase -i main`. After running this command you can type `i` key in order to have an ability to edit all of your commits. Choose your last commits that broke your PR and type the `d` flag near them. It means that you'll drop them from your PR. After doing it, press `i` again. And run `:wq` for saving this configuration.
> More info about the `git rebase` command you can find [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase).
> 
> Let me know if you need any help

Ok, thank you very much! I'll try to do it

**Comment by webdevtoday** [comment]
@MaxKalashnyk Hi! Please, check my PR, I've just done those steps.

**Comment by webdevtoday** [comment]
@MaxKalashnyk I deleted all of my commits. And then I added my files again to project fold
Снимок
Снимок1
Снимок2
Снимок3



er.

**Comment by webdevtoday** [comment]
@MaxKalashnyk But I had a problem at the end. I added screens of my console with my actions. I don't understand, how it happened with me, I did PR-s to homeworks repo many times.

**Comment by MaxKalashnyk** [comment]
> @MaxKalashnyk But I had a problem at the end. I added screens of my console with my actions. I don't understand, how it happened with me, I did PR-s to homeworks repo many times.

I saw your screenshots. You ran all commands properly till the `git push --set-upstream origin memory-pair-game`.
As you have already created this remote upstream before, you didn't need to create it one more time.
I think in that case using `git push origin memory-pair-game` or just `git push` would help.

**Review by MaxKalashnyk — APPROVED** [review]
LGTM!
I see you managed to fixed your irrelevant commits.
I think we can approve this work. I hope you can do more complex and intersting designs of your apps in future. Now it's quite simple :) 
But it's no big deal. You did a great job!

**Comment by webdevtoday** [comment]
@MaxKalashnyk Thank you! Thank you!Thank you!!!! Thank you very much! You're really help me!!

**Pull request merged**